exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Configuration.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.service;
+
+import com.exonum.binding.core.storage.database.Fork;
+import com.google.protobuf.Message;
+
+/**
+ * Configuration parameters of Exonum service. Network administrators agree on and pass
+ * the configuration parameters as a service-specific protobuf message in the transactions
+ * which add that service instance to the network. After Exonum starts the service, it
+ * {@linkplain Service#configure(Fork, Configuration) passes the configuration parameters}
+ * to the newly created service instance.
+ *
+ * <p>Service reconfiguration is not currently supported, but will be added soon.
+ *
+ * @see Service#configure(Fork, Configuration)
+ */
+public interface Configuration {
```

Comment:
Shouldn't this be `ServiceConfiguration`? And implementation `ServiceConfigurationImpl`?
It's good as it is as well though, being in `service` package.

Reply:
I went with a shorter name (`Configuration`) inside `service` package, so that `Service` is not repeated where configuration is used in the Service#configure signature. 

`ServiceConfiguration` name can probably be improved, but I don't currently see anything particularly specific about that implementation.